libraries/rush-lib/src/logic/operations/OperationExecutionManager.ts

```diff
@@ -216,11 +220,14 @@ export class OperationExecutionManager {
   /**
    * Handles the result of the operation and propagates any relevant effects.
    */
-  private _onOperationComplete(record: OperationExecutionRecord): void {
+  private _onOperationComplete(record: OperationExecutionRecord, executionQueue: AsyncOperationQueue): void {
     const { runner, name, status } = record;
 
-    let blockCacheWrite: boolean = !runner.isCacheWriteAllowed;
-    let blockSkip: boolean = !runner.isSkipAllowed;
+    const buildCacheContext: IOperationBuildCacheContext | undefined =
```

Comment:
If OperationExecutionManager takes in its constructor (or in executeAsync) the PhasedOperationHooks object, we can put the `beforeExecuteOperation` and `afterExecuteOperation` hooks on said lifecycle object and make them available to other plugins. Then all knowledge of build cache state can be confined to a single plugin instance that taps those hooks, and the mapping between OperationExecutionRecord and build cache state can be a map captured in the closure of the `apply()` method of said plugin, rather than a global singleton on the class.

Reply:
That's a good idea. Work on it.